Model/Validation/PreventAdminActions.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+namespace Paytrail\PaymentService\Model\Validation;
+
+use Magento\Customer\Model\Session;
+
+class PreventAdminActions
+{
+    const KEY_LOGIN_AS_CUSTOMER_SESSION = 'logged_as_customer_admind_id';
+
+    /**
+     * @var Session
+     */
+    private Session $customerSession;
+
+    /**
+     * @param Session $customerSession
+     */
+    public function __construct(
+        Session $customerSession
```

Comment:
Could you add a proxy declaration for this in di.xml?
We only want the load the session into this class when we require the information. Not during callstack initialization.
Example from: module-customer/etc/di.xml:
```
    <type name="Magento\Customer\Model\Authorization\CustomerSessionUserContext">
        <arguments>
            <argument name="customerSession" xsi:type="object">Magento\Customer\Model\Session\Proxy</argument>
        </arguments>
    </type>
```

Reply:
Done